appserver/web/web-glue/src/main/java/com/sun/enterprise/web/VirtualServer.java

```diff
@@ -920,109 +925,55 @@ synchronized void setLogFile(String logFile, String logLevel, String logServiceF
-            if (newLogger == null) {
-                newLogger = new Logger(lname, null) {
-                    // set thread id, see LogDomains.getLogger method
-                    @Override
-                    public void log(LogRecord record) {
-                        if (record.getResourceBundle() == null) {
-                            ResourceBundle bundle = getResourceBundle();
-                            if (bundle != null) {
-                                record.setResourceBundle(bundle);
-                            }
-                        }
-                        record.setThreadID((int) Thread.currentThread().getId());
-                        super.log(record);
-                    }
-
-                    // use the same resource bundle as default vs logger
-                    @Override
-                    public ResourceBundle getResourceBundle() {
-                        return rb;
-                    }
-
-                    @Override
-                    public synchronized void addHandler(Handler handler) {
-                        super.addHandler(handler);
-                        if (handler instanceof FileLoggerHandler) {
-                            ((FileLoggerHandler) handler).associate();
-                        }
-                    }
-
-                    @Override
-                    public synchronized void removeHandler(Handler handler) {
-                        if (!(handler instanceof FileLoggerHandler)) {
-                            super.removeHandler(handler);
-                        } else {
-                            boolean hasHandler = false;
-                            Handler[] hs = getHandlers();
-                            if (hs != null) {
-                                for (Handler h : hs) {
-                                    if (h == handler) {
-                                        hasHandler = true;
-                                        break;
-                                    }
-                                }
-                            }
-                            if (hasHandler) {
-                                super.removeHandler(handler);
-                                ((FileLoggerHandler) handler).disassociate();
-                            }
-                        }
-                    }
-                };
-
-                synchronized (Logger.class) {
-                    LogManager.getLogManager().addLogger(newLogger);
-                }
-            }
+            setLogger(newLogger);
+            return;
+        }
 
-            // remove old handlers if necessary
-            Handler[] handlers = newLogger.getHandlers();
-            if (handlers != null) {
-                for (Handler h : handlers) {
-                    newLogger.removeHandler(h);
-                }
+        // remove old handlers if necessary
+        // FIXME: does not respect the configuration - why?
```

Comment:
Fixme?

Reply:
> TestContainers things are duplicated from the TestContainers PR. These should be removed.
> 
> JUnit dependency is duplicated in 100s of POM files. It used to be in the parent POM and needs to be there again

- yes, that will go away after the merge of the prerequisity
- junit dependency is a thing of P4176 too, so I will respond there.